src/Networking/NexusMods.Networking.HttpDownloader/DTOs/DownloadState.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Text.Json.Serialization;
+using NexusMods.Paths;
+
+namespace NexusMods.Networking.HttpDownloader.DTOs;
+
+/// <summary>
+/// Overall state of a download
+/// </summary>
+public class DownloadState
```

Comment:
This should probably be an internal API. Any data that we need to expose from here can be done so through an interface.

You can make internals visible in test project with:

```
<ItemGroup>
    <InternalsVisibleTo Include="NexusMods.Networking.HttpDownloader.Tests" />
</ItemGroup>
```

Inside the project's `.csproj`